src/alire/alire-publish.adb

```diff
@@ -0,0 +1,257 @@
+
+      if Deployer.Supports_Hashing then
+         for Kind in Hashes.Kinds loop
+            declare
+               Hash : constant Hashes.Any_Hash :=
+                        Hashes.New_Hash
+                          (Kind,
+                           Deployer.Compute_Hash
+                             (Context.Tmp_Deploy_Dir.Filename, Kind));
+            begin
+               Log_Success ("Computed hash: " & String (Hash));
+               Context.Origin.Add_Hash (Hash);
+            end;
+         end loop;
+      end if;
+
+      --  Pull the repo/unpack source archive
+
+      Deployer.Deploy (Context.Tmp_Deploy_Dir.Filename).Assert;
+
+   end Deploy_Sources;
+
+   -----------------------------
+   -- Generate_Index_Manifest --
+   -----------------------------
+
+   procedure Generate_Index_Manifest (Context : in out Data) with
+     Pre => GNAT.OS_Lib.Is_Directory (Context.Tmp_Deploy_Dir.Filename);
+   --  Bind the user manifest with the origin TOML object and create the index
+   --  manifest.
+
+   procedure Generate_Index_Manifest (Context : in out Data) is
+      use Directories.Operators;
+      User_Manifest : constant Any_Path :=
+                       Context.Tmp_Deploy_Dir.Filename / Roots.Crate_File_Name;
+   begin
+      if not GNAT.OS_Lib.Is_Read_Accessible_File (User_Manifest) then
+         Raise_Checked_Error
+           ("User manifest not found at expected location"
+            & " (${SRC_ROOT}/" & Roots.Crate_File_Name & ").");
+      end if;
+
+      declare
+         use Ada.Text_IO;
+         use TOML;
+         TOML_Manifest  : constant TOML_Value :=
+                            TOML_Load.Load_File (User_Manifest);
+         TOML_Origin    : constant TOML_Value := Create_Table;
+         Name           : constant Crate_Name :=
+                            +TOML_Manifest.Get (TOML_Keys.Name).As_String;
+         Version        : constant Semver.Version :=
+                            Semver.Parse
+                              (TOML_Manifest.Get
+                                 (TOML_Keys.Version).As_String);
+         Index_Manifest : constant Any_Path :=
+                            TOML_Index.Manifest_File (Name, Version);
+         Index_File     : File_Type;
+      begin
+         TOML_Origin.Set (TOML_Keys.Origin, Context.Origin.To_TOML);
+
+         --  Take the user manifest and bundle it under the proper index
+         --  manifest name with the origin we are being provided with:
+
+         Ada.Directories.Copy_File (User_Manifest, Index_Manifest);
+
+         Open (Index_File, Append_File, Index_Manifest);
+         New_Line (Index_File);
+         TOML.File_IO.Dump_To_File (TOML_Origin, Index_File);
+         Close (Index_File);
+
+         Log_Success
+           ("Your index manifest file has been generated at "
+            & TTY.URL (Index_Manifest));
+         Log_Success
+           ("Please place it at " & TTY.URL (TOML_Index.Manifest_Path (Name))
```

Comment:
Here we can use the GitHub upload link, for instance:
```
Upload the manifest using this link:
https://github.com/alire-project/alire-index/upload/devel-0.4/index/li/libhello/
```

Reply:
Neat, I see it creates the PR and it seems to work even if the directory doesn't exist yet.

Reply:
Yep, it's perfect for us :)